src/onedrive.d

```diff
@@ -629,8 +630,57 @@ final class OneDriveApi
 		try {
 			http.perform();
 		} catch (CurlException e) {
-			// Potentially Timeout was reached on handle error
-			log.error("ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?");
+			// Parse and display error message received from OneDrive
+			log.error("ERROR: OneDrive returned an error with the following message:");
+			auto errorArray = splitLines(e.msg);
+			string errorMessage = errorArray[0];
+			
+			if (canFind(errorMessage, "Couldn't connect to server on handle")) {
+				// This is a curl timeout
+				log.error("  Error Message: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?");
```

Comment:
Hmm, having these error messages every time sounds a bit disturbing for the log file. Imagine travelling on an airplane, you are of course offline, so having these error messages might be a bit painful.

What I think would be great are some messages after 1/5/10min or so... Not sure so how exactly.

Reply:
That's why the sleep is inside the loop, to slow down those error messages.

Regarding the airplane scenario - that's exactly why I raised it.
$ sudo zgrep -E -ci "onedrive.*internet" /var/log/syslog.7.gz
25419305

That's one day's worth of logging.  Regarding the "high CPU" - it's "relatively" high.  onedrive 2.3.8 goes up from 0.5% of my CPU to 5% when it goes into the unhandled loop - it becomes the highest power drain on my laptop according to powertop when I'm on the airplane.  I'm travelling the next 2 weeks, so i can test daily, but it'll be 10:00 UTC before I'm able to test again.  Because @abraunegg is updating the PR, I can just keep fetching pr694 as before to get those updates, right?

Cheers, and thanks for the super fast collaboration!

Reply:
@norbusan 
> What I think would be great are some messages after 1/5/10min or so... Not sure so how exactly.

Because of this: `Thread.sleep(dur!"seconds"(retryAttempts*backoffInterval));`  log output should slow dramatically 

@docsmooth
> Because @abraunegg is updating the PR, I can just keep fetching pr694 as before to get those updates, right?

Correct - you can keep re-pulling this PR and get all the latest changes

>I'm travelling the next 2 weeks
Any real-world testing / feedback is greatly appreciated

Reply:
Hmm, I wouldn't say that is slows down dramatically, since it is a linear progression 2,4,6,8,10,12,14,16secs. If that would be an exponential progression 2,4,8,16,32,64,... that would be fine I guess.

Reply:
@norbusan 
Latest update will do the following as interval:
* 2
* 6
* 12
* 20
* 30
* 42
* 56
...
* 3660

3660 is hit after ~60 cycles which is akin to 1260 minutes, or after ~21 hrs of Internet unavailability. The service will then re-try every 3600 seconds (1hr) for a further 9940 cycles before giving up.

Reply:
Ah, ` n*(n-1)` ;-) You could have done it easier with `retryAttempts ** 2` and get 1,4,9,16,25,... ;-) Probably makes the code easier to read.